### ISSUE TITLE
Unsat core not stored in the interpolant at the subsumption point

### ISSUE THREAD

**Issue by rasoolmaghareh** [tracer-x/klee#317]
In a program if both child of a node is subsumed, the interpolant generated for their parents is empty. For example consider the following program:

```
#include <klee/klee.h>

int main() {
  int wcet,q;
  int a,b,c;

  klee_make_symbolic(&a, sizeof(int), "a");
  klee_make_symbolic(&b, sizeof(int), "b");
  klee_make_symbolic(&c, sizeof(int), "c");
  klee_make_symbolic(&wcet, sizeof(int), "wcet");

  klee_assume(wcet == 0);
  q = 5;

  if (a >= 0){
     wcet +=5;        \\PP 9
     q +=3;
  }else{
     wcet+=5;         \\PP 2
     q +=4;
  }
  
  if (b != 1){
     wcet+=2;
     q +=5;
  }else{
     wcet+=2;
     q +=6;
  }
  
  if (c >= 0){
     wcet+=2;
     q +=5;
  }else{
     wcet+=2;
     q +=6;
  }

  klee_assert(wcet < 10);
} 
```

The subsumption table entry at PP 2 (after the `a<0`) is:
```
Program point = 41962112
interpolant = (Eq 0
     (ReadLSB w32 0 __shadow__wcet))
concretely-addressed store = [
        address:
                function/value: main/  %wcet = alloca i32, align 4
                stack: (empty)
                offset: 0
        content:
                function/value: main/  %add = add nsw i32 %6, 5, !dbg !136
                (Add w32 5
          (ReadLSB w32 0 __shadow__wcet))
                reason(s) for storage:
                        branch infeasibility [main: Line 40]
]
symbolically-addressed store = []
concretely-addressed historical store = []
symbolically-addressed historical store = []
existentials = [__shadow__wcet]
```

At PP9 (after the `a>=0`), both children are subsumed and the interpolant in the subsumption table entry is empty:
```
Program point = 41961152
interpolant = (empty)
concretely-addressed store = [
        address:
                function/value: main/  %wcet = alloca i32, align 4
                stack: (empty)
                offset: 0
        content:
                function/value: main/  %add = add nsw i32 %6, 5, !dbg !136
                (Add w32 5
          (ReadLSB w32 0 __shadow__wcet))
                reason(s) for storage:
                        branch infeasibility [main: Line 40]
]
symbolically-addressed store = []
concretely-addressed historical store = []
symbolically-addressed historical store = []
existentials = [__shadow__wcet]
```

While is should contain: ` (Eq 0 (ReadLSB w32 0 __shadow__wcet))`

**Comment by domainexpert** [comment]
@rasoolmaghareh This seems to be a bug, where in `SubsumptionTableEntry::subsumed()`, whenever subsumption is decided by the pre-solver without calling the solver, the unsat core is not computed, and therefore the unsat core interpolation is not done.

**Comment by rasoolmaghareh** [comment]
Thanks for your review. I think one fix is that to pass the interpolant stored in the subsumption table entry as the unsat core and then perform unsat core interpolation.

**Comment by domainexpert** [comment]
Resolved via #320.
